lib/galaxy/model/__init__.py

```diff
@@ -732,7 +732,7 @@ class User(Base, Dictifiable, RepresentById):
     create_time: Mapped[datetime] = mapped_column(default=now, nullable=True)
     update_time: Mapped[datetime] = mapped_column(default=now, onupdate=now, nullable=True)
     email: Mapped[str] = mapped_column(TrimmedString(255), index=True)
-    username: Mapped[Optional[str]] = mapped_column(TrimmedString(255), index=True, unique=True)
+    username: Mapped[str] = mapped_column(TrimmedString(255), index=True, unique=True)
```

Comment:
If I understood correctly the third bullet point in https://github.com/galaxyproject/galaxy/pull/17958, we need to add `nullable=True` to `mapped_column` to reflect the real table definition.

The same applies to the rest of the model attributes that changed to non-optional type hints here.

Reply:
Hehe, I was looking for that, it would have saved me from changing all those unit tests 😆